src/Classes/TreeTab.lua

```diff
@@ -379,8 +379,26 @@ function TreeTabClass:OpenSpecManagePopup()
 end
 
 function TreeTabClass:OpenImportPopup()
+	local versionLookup = "tree/([0-9]+)%.([0-9]+)%.([0-9]+)/"
 	local controls = { }
-	local function decodeTreeLink(treeLink)
+	local function decodeTreeLink(treeLink, newTreeVersion)
+			-- newTreeVersion is passed in as an output of validateTreeVersion(). It will always be a valid tree version text string
+			-- If there was a version on the url, and it changed the version of the active spec, dump the active spec and get one of the right version. 
+		if newTreeVersion ~= self.specList[self.activeSpec].treeVersion then
+			local newSpec = new("PassiveSpec", self.build, newTreeVersion)
+			newSpec:SelectClass(self.build.spec.curClassId)
+			newSpec:SelectAscendClass(self.build.spec.curAscendClassId)
+			newSpec.title = self.build.spec.title
+			oldSpec = self.specList[self.activeSpec]
+			self.specList[self.activeSpec] = newSpec
+			-- only wipe after swapping out
+			oldSpec = {}
```

Comment:
I don't think wiping the old spec is necessary.  It should get cleaned up by the garbage collector